### PR TITLE
Add API documentation [ST-2137]

### PR DESCRIPTION
**Pull request by tdphillips**
The doc is pretty much good to go. We're just waiting on opinions regarding table ordering.

These are a little ugly but they're functional. I have a few minor edits but these are basically what's in Drive at the moment.

### CODE REVIEW

File: doc/source/developers/api/basic_usage.rst [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+
+        **frequency**
+
+        The frequency with which the saved search will be sent. Currently, the API only supports daily frequencies.
+
+        **new_search**
+
+        Returns true if this is a new search and false if the saved search already exists. My.jobs will not allow duplicate searches to be setup.
+    * - Example
+      - https://secure.my.jobs/api/v1/savedsearch/?callback=finish_ss&email=pat@email.com&url=http%3A%2F%2Fwww.my.jobs%2Fjobs%2F%3Fq%3Dinternet&username=you@yourcompany.com&api_key=1234567890abcdefghijklmnop
```
Comment (tdphillips) [inline doc/source/developers/api/basic_usage.rst]: This makes for a very wide page.

File: doc/source/developers/api/basic_usage.rst [hunk 2]
```diff
@@ -0,0 +1,120 @@
+**DirectEmployers Association**
```
Comment (aspidites) [inline doc/source/developers/api/basic_usage.rst]: What is this line doing here?

**Comment by aspidites** [comment]
No strong opinion, but should we move examples up top as the first row in each table?

File: doc/source/developers/api/basic_usage.rst (hunk 2 above)
Comment (tdphillips) [inline doc/source/developers/api/basic_usage.rst]: It was in the original doc. ¯\\\_(ツ)\_/¯

**Comment by tdphillips** [comment]
@fezick The document @aspidites is talking about appears to be intended for third parties. What do you think?

File: doc/source/developers/api/basic_usage.rst (hunk 2 above)
Comment (aspidites) [inline doc/source/developers/api/basic_usage.rst]: I'd get rid of it then. Given this is on github.com/DirectEmployers and is generated on directemployers.github.io, I think it's pretty obvious now.

**Comment by darrint** [comment]
I like this. I need to add similar docs for dynamic reports apis. No strong opinion on table ordering.

**Comment by tdphillips** [comment]
There's been no further input one way or the other. Updating now.

**Comment by darrint** [comment]
Warning, treated as error:
/MyJobs/doc/source/developers/api/developer_guide.rst:101: ERROR: Error in "code-block" directive:
1 argument(s) required, 0 supplied.

.. code-block::

    $ ./manage.py shell
    > from myjobs.models import User
    > from tastypie.models import create_api_key
    > user = User.objects.get(email="your_email@example.com")
    > create_api_key(User, instance=user, created=True)

make: *** [html] Error 1

**Comment by tdphillips** [comment]
I can't reproduce.

**Pull request merged**